python/py-tensorflow_macos/Portfile

```diff
@@ -0,0 +1,113 @@
+distfiles           ${github.project}-${github.version}${extract.suffix}:tarball \
+                    ${github.project}-${github.version}${github.tag_suffix}${extract.suffix}:releases
+
+master_sites        ${github.homepage}/tarball/${git.branch}:tarball \
+                    github.master_sites ${github.homepage}/releases/download/${git.branch}:releases
+
+checksums           [lindex [split [lindex ${distfiles} 0] :] 0] \
+                    rmd160  975a2ee2508df2fe801b841d3f6f8a0172854292 \
+                    sha256  623237c06941d26f52280535c0139c9d58fc214e0378b4513803a65bfda48ff6 \
+                    size    8095 \
+                    [lindex [split [lindex ${distfiles} 1] :] 0] \
+                    rmd160  812afc151dd87abb657df6d8e66ad8681c91a3bd \
+                    sha256  80807f8b0fc4a98ffa0b7395304fa61239f246a84f1eedbca70a5a1847d4dead \
+                    size    376699291
+
+extract.cmd         ${portutil::autoconf::tar_command}
+extract.pre_args    -xzf
+extract.post_args
+
+if {${os.platform} eq "darwin" && ${os.major} < 20} {
+    known_fail      yes
+    pre-fetch {
+        ui_error "TensorFlow with ML Compute acceleration is only available \
+            on macOS 11.0 and later."
+        error {unsupported platform}
+    }
+}
+
+python.versions 38
+
+if {${name} ne ${subport}} {
+    conflicts       py${python.version}-tensorflow \
+                    py${python.version}-tensorflow-addons
+
+    depends_build-append \
+                    port:py${python.version}-pip \
+                    port:py${python.version}-wheel
+
+    depends_run-append \
+                    port:py${python.version}-absl \
+                    port:py${python.version}-astunparse \
+                    port:py${python.version}-flatbuffers \
+                    port:py${python.version}-gast \
+                    port:py${python.version}-grpcio \
+                    port:py${python.version}-h5py \
+                    port:py${python.version}-keras_preprocessing \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-opt_einsum \
+                    port:py${python.version}-protobuf3 \
+                    port:py${python.version}-tensorflow_estimator \
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-tensorboard \
+                    port:py${python.version}-termcolor \
+                    port:py${python.version}-typing_extensions \
+                    port:py${python.version}-wrapt \
+                    port:py${python.version}-typeguard
+    
+    use_configure   no
+
+    build {}
+
+    destroot.cmd    pip-${python.branch}
+    destroot.args   --ignore-installed \
+                    --no-cache-dir \
+                    --no-dependencies \
+                    --root ${destroot}
+    destroot.post_args
+
+    destroot {
+        foreach whl [glob -type f ${workpath}/${github.project}/x86_64/tensorflow*cp${python.version}*.whl] {
```

Comment:
Re: https://github.com/macports/macports-ports/pull/10618#issuecomment-815871065

> Not really sure exactly what you are asking ? Once we have Apple Silicon machines back in the buildbots we can just see what that makes of this port. Until then you could ask on the devel mailing if anyone there with a M1 machine could give it a try.

This line. For M1, the `x86_64` directory must be replaced with the M1 one, and the M1 `.whl`'s will be used.

Reply:
Right, so then you should use ${os.arch} to determine the architecture and adjust this accordingly.

Reply:
Correct. Make that ${build_arch}. See e.g.

https://github.com/macports/macports-ports/blob/master/lang/gcc-devel/Portfile

For an example of a port that does this to do something different on apple silicon machines.

Reply:
Thanks, done.